src/Data/OpenApi/Internal.hs

```diff
@@ -1425,6 +1445,23 @@ instance ToJSON Callback where
 -- Manual FromJSON instances
 -- =======================================================================
 
+instance FromJSON OpenApiSpecVersion where
+  parseJSON = withText "OpenApiSpecVersion" $ \str ->
+            let validatedVersion :: Either String Version
+                validatedVersion = do
+                  parsedVersion <- readVersion str 
+                  upperBound <- readVersion "3.0.3" -- Latest known version that works with the spec
+                  lowerBound <- readVersion "3.0.0"
+                  unless ((parsedVersion >= lowerBound) && (parsedVersion <= upperBound)) $ Left ("The provided version " <> showVersion parsedVersion <> " is out of the allowed range >=3.0.0 && <=3.0.3")
```

Comment:
```suggestion
                  unless ((parsedVersion >= lowerBound) && (parsedVersion <= upperBound)) $
                      Left ("The provided version " <> showVersion parsedVersion <> " is out of the allowed range >=3.0.0 && <=3.0.3")
```

also, can you please use `showVersion` with the bounds you set in the error message?